crates/engine/tree/src/tree/mod.rs

```diff
@@ -342,6 +368,69 @@ where
                     }
                 }
             }
+
+            if self.should_persist() && !self.persistence_state.in_progress {
+                let blocks_to_persist = self.get_blocks_to_persist();
+                let (tx, rx) = oneshot::channel();
+                self.persistence.save_blocks(blocks_to_persist, tx);
+                self.persistence_state.in_progress = true;
+                self.persistence_state.rx = Some(rx);
```

Comment:
do we need the extra bool, if in progress is `Some` ?

Reply:
makes sense, removed and added some methods to `PersistenceState` to make it easier to interact with it